### PR TITLE
APP-1447: Integration Bridge needs support proxy server to connect to POD

### PR DESCRIPTION
**Pull request by Hamitay-Daitan**
Added a proxy model and field to Integration Properties.
Fixed unit tests and added new ones based on the proxy model.

### CODE REVIEW

**Review by crepache — APPROVED** [review]
LGTM

File: integration-api-client/src/main/java/org/symphonyoss/integration/api/client/AppAuthenticationProxyApiClient.java [hunk 1]
```diff
@@ -33,15 +33,20 @@
 
   private AppAuthenticationProxy proxy;
 
-  public AppAuthenticationProxyApiClient(EntitySerializer serializer, AppAuthenticationProxy proxy) {
+  String serviceName;
```
Comment (robson-sanchez) [inline integration-api-client/src/main/java/org/symphonyoss/integration/api/client/AppAuthenticationProxyApiClient.java]: It should be a private field.

File: integration-api-client/src/test/java/org/symphonyoss/integration/api/client/AppAuthenticationProxyApiClientTest.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
+import org.slf4j.LoggerFactory;
```
Comment (robson-sanchez) [inline integration-api-client/src/test/java/org/symphonyoss/integration/api/client/AppAuthenticationProxyApiClientTest.java]: This import is not required.

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java [hunk 3]
```diff
@@ -160,7 +160,7 @@ public void onCreate(String integrationUser) {
 
     try {
       authenticate(integrationUser);
-      updateConfiguration(integrationUser);
+        updateConfiguration(integrationUser);
```
Comment (robson-sanchez) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java]: Blank space?

File: integration-api-client/src/main/java/org/symphonyoss/integration/api/client/SymphonyApiClient.java [hunk 4]
```diff
@@ -1,12 +1,15 @@
 package org.symphonyoss.integration.api.client;
 
+import org.glassfish.jersey.client.ClientProperties;
```
Comment (robson-sanchez) [inline integration-api-client/src/main/java/org/symphonyoss/integration/api/client/SymphonyApiClient.java]: This import is not required.

File: integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/ConnectionInfo.java [hunk 5]
```diff
@@ -53,11 +55,20 @@ public void setMinVersion(String minVersion) {
     this.minVersion = minVersion;
   }
 
+  public void setProxyUser(String proxyUser) { this.proxy.setUser(proxyUser); }
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/ConnectionInfo.java]: Why do you set proxy info directly in this object?

File: integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/IntegrationProperties.java [hunk 6]
```diff
@@ -20,8 +20,10 @@
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.context.annotation.Configuration;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/IntegrationProperties.java]: These imports are not being used.

File: integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/IntegrationProperties.java [hunk 7]
```diff
@@ -314,6 +333,10 @@ public String getIntegrationBridgeUrl() {
     return String.format("https://%s:%s/%s", integrationBridge.getHost(), port, INTEGRATION_CONTEXT);
   }
 
+  public Map<String, ConnectionInfo> getServices() { return services;  }
+
+  public void setServices(Map<String, ConnectionInfo> services) { this.services = services; }
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/IntegrationProperties.java]: Should you override those service externally?

File: integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/IntegrationProperties.java [hunk 8]
```diff
@@ -314,6 +333,10 @@ public String getIntegrationBridgeUrl() {
     return String.format("https://%s:%s/%s", integrationBridge.getHost(), port, INTEGRATION_CONTEXT);
   }
 
+  public Map<String, ConnectionInfo> getServices() { return services;  }
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/IntegrationProperties.java]: Should you export those services as a Map object? It should be better this method returns a list of services or you might provide a method to get service by id.

File: integration-commons/src/test/java/org/symphonyoss/integration/model/yaml/ConnectionInfoTest.java [hunk 9]
```diff
@@ -3,6 +3,8 @@
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.net.URI;
```
Comment (robson-sanchez) [inline integration-commons/src/test/java/org/symphonyoss/integration/model/yaml/ConnectionInfoTest.java]: This import is not used.

File: integration-commons/src/test/java/org/symphonyoss/integration/model/yaml/ProxyConnectionInfoTest.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+      "ProxyConnectionInfo{uri='URI', user=user', password=password'}";
+
+  @Test
+  public void testProxyConnectionInfo() {
+    ProxyConnectionInfo proxy = new ProxyConnectionInfo();
+    proxy.setUser(USER);
+    proxy.setPassword(PASSWORD);
+    proxy.setURI(URI);
+
+    Assert.assertEquals(USER, proxy.getUser());
```
Comment (robson-sanchez) [inline integration-commons/src/test/java/org/symphonyoss/integration/model/yaml/ProxyConnectionInfoTest.java]: Please use the static import.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**